nav2_world_model/include/nav2_world_model/costmap_representation.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_WORLD_MODEL__COSTMAP_REPRESENTATION_HPP_
+#define NAV2_WORLD_MODEL__COSTMAP_REPRESENTATION_HPP_
+
+#include <string>
+#include <memory>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_world_model/world_representation.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "tf2_ros/transform_listener.h"
+#include "nav2_util/point2d.hpp"
+#include "nav2_world_model/region_visualizer.hpp"
+
+namespace nav2_world_model
+{
+
+class CostmapRepresentation : public WorldRepresentation
```

Comment:
Does this inheritance structure still make sense when the World Model consists of plugins?

Reply:
To be addressed in #565.